lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -251,24 +253,29 @@ private static function get_inner_blocks( $attributes, $block ) {
 		}
 
 		// Load inner blocks from the navigation post.
-		if ( array_key_exists( 'ref', $attributes ) ) {
+		if ( $navigation_post_id ) {
 			$inner_blocks = static::get_inner_blocks_from_navigation_post( $attributes );
 		}
 
 		// If there are no inner blocks then fallback to rendering an appropriate fallback.
 		if ( empty( $inner_blocks ) ) {
-			$inner_blocks = static::get_inner_blocks_from_fallback( $attributes );
+			$inner_blocks            = static::get_inner_blocks_from_fallback( $attributes );
+			$fallback_naviation_post = gutenberg_block_core_navigation_get_fallback_post();
+
+			if ( $fallback_naviation_post ) {
+				$navigation_post_id = $fallback_naviation_post->ID;
+			}
 		}
 
 		/**
 		 * Filter navigation block $inner_blocks.
 		 * Allows modification of a navigation block menu items.
 		 *
 		 * @since 6.1.0
-		 *
+		 * @param int|null $navigation_post_id The navigation post ID.
 		 * @param \WP_Block_List $inner_blocks
 		 */
-		$inner_blocks = apply_filters( 'block_core_navigation_render_inner_blocks', $inner_blocks );
+		$inner_blocks = apply_filters( 'block_core_navigation_render_inner_blocks', $inner_blocks, $navigation_post_id );
```

Comment:
In terms of separation on concerns, it'd be great if we could avoid passing the `$navigation_post_id` to the `block_core_navigation_render_inner_blocks` filter (which in an ideal world should probably be ignorant of post IDs), but alas, I can't seem to see any way to do so.

Reply:
It might be wise to avoid coupling to the post ID and instead pass the `ref` from the block. Why? Because in the future we may wish to move to using Post slugs to reference the `wp_navigation`.

Reply:
@ockham I'll give this some thought and see if there's another way to approach this.

@getdave I do agree, and I can look at refactoring this so it's the case if possible. That said though there are already some assumptions in this code that the `ref` value is a Post ID anyway, for example [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php#L186) and [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php#L234-L236). If it's not possible to remove this assumption on this iteration, would you consider it a blocker?

Reply:
>I can't seem to see any way to do so.

@ockham me neither, nothing that stands out anyway 😕

Reply:
The existing assumptions exist because they exist. But we shouldn't compound them if we can avoid it.

Reply:
@ockham thanks for this idea. I've implemented it [here](https://github.com/WordPress/gutenberg/pull/57754/commits/7872834f398ff2d5a0136a22e8b400b748a0094c) and it works well from my testing.

One thing worth noting is that inside `get_inner_blocks_from_fallback` two functions run:

* `gutenberg_block_core_navigation_get_fallback_post`
* `gutenberg_block_core_navigation_get_fallback_blocks`

Both of these make a DB query for the same post. I could pass the `$fallback_navigation_post` object as a param into `gutenberg_block_core_navigation_get_fallback_blocks` which would solve this, but I'm unsure on whether changing the signature of the function is acceptable?

Reply:
> One thing worth noting is that inside `get_inner_blocks_from_fallback` two functions run:
> 
> * `gutenberg_block_core_navigation_get_fallback_post`
> * `gutenberg_block_core_navigation_get_fallback_blocks`
>
> Both of these make a DB query for the same post. I could pass the `$fallback_navigation_post` object as a param into `gutenberg_block_core_navigation_get_fallback_blocks` which would solve this, but I'm unsure on whether changing the signature of the function is acceptable?

TBH I think I'd roll back the changes to `get_inner_blocks_from_fallback` that we've made as part of this PR and would get rid of the `gutenberg_block_core_navigation_get_fallback_post` helper altogether. Instead, I'd call `gutenberg_hook_first_last_children_into_core_navigation()` from `block_core_navigation_get_fallback_blocks()` (which I believe gets its `gutenberg_` prefix during the build process) in `packages/block-library/src/navigation/index.php`. That function has both a `$navigation_post` object and parsed blocks readily available 🙂

Reply:
Good suggestion! I've updated that here https://github.com/WordPress/gutenberg/pull/57754/commits/0019afa6b79724ab55e3f371b7bb1447350dc835